File: app/src/main/java/net/nashihara/naroureader/views/adapters/NovelBodyFragmentViewPagerAdapter.java
package net.nashihara.naroureader.views.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import net.nashihara.naroureader.fragments.NovelBodyFragment;

public class NovelBodyFragmentViewPagerAdapter extends FragmentStatePagerAdapter {
    private static final String TAG = NovelBodyFragmentViewPagerAdapter.class.getSimpleName();

    private String ncode;
    private String title;
    private int totalPage;

    public NovelBodyFragmentViewPagerAdapter(FragmentManager fm, String ncode, String title, int totalPage) {
        super(fm);
        this.ncode = ncode;
        this.title = title;
        this.totalPage = totalPage;
    }

    @Override
    public Fragment getItem(int position) {
        return NovelBodyFragment.newInstance(ncode, title, "", position +1, totalPage);
    }

    @Override
    public int getCount() {
        return totalPage;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return title;
    }
}

File: app/src/main/java/net/nashihara/naroureader/adapters/NovelBodyFragmentViewPagerAdapter.kt
package net.nashihara.naroureader.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

import net.nashihara.naroureader.fragments.NovelBodyFragment

class NovelBodyFragmentViewPagerAdapter(
        fm: FragmentManager,
        private val ncode: String,
        private val title: String,
        private val totalPage: Int) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return NovelBodyFragment.newInstance(ncode, title, "", position + 1, totalPage)
    }

    override fun getCount(): Int {
        return totalPage
    }

    override fun getPageTitle(position: Int): CharSequence {
        return title ?: ""
    }
}
